test/integration/tape-sauce.js

```diff
@@ -20,9 +19,15 @@ test('tape - sauce', function (t) {
 
   var zuul = Zuul(config)
 
-  getBrowsers(function (err, allBrowsers) {
-    var browsers = flattenBrowser(browsersToTest, allBrowsers)
-    browsers.forEach(zuul.browser.bind(zuul))
-    verify(t, zuul, err)
+  sauceBrowsers(browsersToTest, function (err, browsers) {
+    t.error(err, 'no error')
```

Comment:
Build is failing because we have a `t.plan()` in `verify()`, so we can't add additional assertions here (`t.error`). We should refactor this later. For now let's change it to:

```js
function (err, browsers) {
  if (err) {
     t.fail(err.message)
     return t.end()
  }

  // ..
}
```

Reply:
@ralphtheninja WDYT?

Reply:
Ya noticed, was planning to pass the error to verify as it was, e.g.

```diff
diff --git a/test/integration/tape-sauce.js b/test/integration/tape-sauce.js
index 5b4fb0b..e59f5a9 100644
--- a/test/integration/tape-sauce.js
+++ b/test/integration/tape-sauce.js
@@ -20,14 +20,13 @@ test('tape - sauce', function (t) {
   var zuul = Zuul(config)
 
   sauceBrowsers(browsersToTest, function (err, browsers) {
-    t.error(err, 'no error')
-    browsers.forEach(function (info) {
+    (browsers || []).forEach(function (info) {
       zuul.browser({
         name: info.api_name,
         version: info.short_version,
         platform: info.os
       })
     })
-    verify(t, zuul)
+    verify(t, zuul, err)
   })
 })
```

Reply:
Ah, sure, that would also work.

Reply:
Or

```js
diff --git a/test/integration/tape-sauce.js b/test/integration/tape-sauce.js
index 5b4fb0b..267cb84 100644
--- a/test/integration/tape-sauce.js
+++ b/test/integration/tape-sauce.js
@@ -28,6 +28,6 @@ test('tape - sauce', function (t) {
         platform: info.os
       })
     })
-    verify(t, zuul)
+    verify(t, zuul, 1)
   })
 })
diff --git a/test/integration/verify-common.js b/test/integration/verify-common.js
index 9f485c1..33fd92c 100644
--- a/test/integration/verify-common.js
+++ b/test/integration/verify-common.js
@@ -1,7 +1,6 @@
-module.exports = function (t, zuul, err) {
+module.exports = function (t, zuul, n) {
   var count = zuul._browsers.length || 1
-  t.plan(count * 8 + 3)
-  t.error(err, 'no error')
+  t.plan(count * 8 + 2 + (n | 0))
 
   zuul.on('browser', function (browser) {
     var consoleOutput = []
```

as the error is only used in `tape-sauce.js`. Whatever you think it's better.

Reply:
I personally prefer the `t.fail` approach of https://github.com/airtap/airtap/pull/126#discussion_r173656660 because if there's an error,  `browsers` will be undefined, and there is no work for `verify`. In a sense the error is "external" to `verify` so I think it should be checked beforehand. Code deduplication isn't always better :)

That said, no solution is perfect and we already had this code, so it's not a problem for this PR to solve.

Reply:
Agreed, will amend.